base/datatype_reference.bal

```diff
@@ -78,3 +78,71 @@ public type Reference record {|
     Identifier identifier?;
     string display?;
 |};
+
+# Create a relative Reference to FHIR Resource.
+#
+# + referenceResourceModel - FHIR Resource model of the Reference
+# + relativePathToReference - Relative path to Reference
+# + return - FHIRError if the Reference resource type is not defined, else FHIR Reference
+public function createRelativeFhirReference(typedesc<anydata> referenceResourceModel, string relativePathToReference) returns Reference|FHIRError {
+    ResourceDefinitionRecord? def = referenceResourceModel.@ResourceDefinition;
+    string? resourceType = def?.resourceType;
+
+    if resourceType is string {
+        return {
+            reference: string `${resourceType}/${relativePathToReference}`,
+            'type: resourceType
+        };
+    } else {
+        return createFHIRError("Resource type of the Reference resource model is not defined", ERROR, PROCESSING, errorType = PROCESSING_ERROR);
+    }
+}
+
+# Create an absolute Reference to FHIR Resource.
+#
+# + referenceResourceModel - FHIR Resource model of the Reference
+# + absolutePathToReference - Absolute path to Reference
+# + return - FHIRError if the Reference resource type is not defined, else FHIR Reference
+public function createAbsoluteFhirReference(typedesc<anydata> referenceResourceModel, string absolutePathToReference) returns Reference|FHIRError {
+    ResourceDefinitionRecord? def = referenceResourceModel.@ResourceDefinition;
+    string? resourceType = def?.resourceType;
+
+    if resourceType is string {
+        return {
+            reference: absolutePathToReference,
+            'type: resourceType
+        };
+    } else {
+        return createFHIRError("Resource type of the Reference resource model is not defined", ERROR, PROCESSING, errorType = PROCESSING_ERROR);
+    }
+}
+
+# Create a contained Reference to FHIR Resource.
+#
+# + targetResource - FHIR Resource which the Reference is to be contained/embedded
+# + resourceTypeReference - Resource typed Reference to be contained/embedded
+# + return - FHIRError if the target FHIR Resource is not a valid FHIR resource, else FHIR Reference
+public function createContainedFhirReference(map<anydata> targetResource, Resource resourceTypeReference) returns Reference|FHIRError {
+    string resourceType = resourceTypeReference.resourceType;
+    anydata? targetContained = targetResource["contained"];
```

Comment:
`anydata` contains `()`

Reply:
Code removed due to non-relevant code in https://github.com/ballerina-platform/module-ballerinax-health.fhir.r4/commit/817dadb5178bde7d809333a26f465c6688d3879a